src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingExerciseIntegrationTestService.java

```diff
@@ -2164,6 +2169,27 @@ void test_redirectGetTemplateRepositoryFilesWithContent(BiFunction<ProgrammingEx
     }
 
     private long getMaxProgrammingExerciseId() {
-        return programmingExerciseRepository.findAll().stream().mapToLong(ProgrammingExercise::getId).max().orElse(1L);
+        return programmingExerciseRepository.findAll(PageRequest.of(0, 1, Sort.by(Sort.Direction.DESC, "id"))).stream().mapToLong(ProgrammingExercise::getId).max().orElse(1L);
```

Comment:
If the query results are already sorted then the largest ID will be first in the stream, right?
```suggestion
        return programmingExerciseRepository.findAll(PageRequest.of(0, 1, Sort.by(Sort.Direction.DESC, "id"))).stream().limit(1).findFirst().map(ProgrammingExercise::getId).orElse(1L);
```
This is still not as good as having a dedicated query to pull the largest ID, but perhaps that is being avoided for a good reason, I don't know.